sdk/consumption/Azure.ResourceManager.Consumption/tests/Scenario/ReservationConsumptionTests.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.TestFramework;
+using Azure.ResourceManager.Consumption.Models;
+using NUnit.Framework;
+
+namespace Azure.ResourceManager.Consumption.Tests
+{
+    internal class ReservationConsumptionTests : ConsumptionManagementTestBase
+    {
+        protected ResourceIdentifier _reservationOrderId;
+        protected ResourceIdentifier _reservationId;
+        protected ResourceIdentifier _scope;
+        private const string _filter = "properties/usageDate ge 2023-04-01 AND properties/usageDate le 2023-12-01";
+
+        public ReservationConsumptionTests(bool isAsync) : base(isAsync)
+        {
+        }
+
+        [SetUp]
+        public void SetUp()
+        {
+            _reservationId = new ResourceIdentifier("<reservation-id>");
+            _reservationOrderId = new ResourceIdentifier("<reservation-id>");
+            _scope = new ResourceIdentifier($"/providers/Microsoft.Capacity/reservationorders/{_reservationOrderId}/reservations/{_reservationId}");
+        }
+
+        [RecordedTest]
+        [Ignore("Current tenants have not reservation resource")]
+        public async Task GetReservationDetails()
+        {
+            ReservationConsumptionResource reservation = Client.GetReservationConsumptionResource(_scope);
+            var list = await reservation.GetReservationDetailsAsync(_filter).ToEnumerableAsync();
```

Comment:
```suggestion
            var first = await reservation.GetReservationDetailsAsync(_filter).FirstOrDefaultAsync(x => true);
```
Since that we only want to get the first element in the test.